### PR TITLE
feat(client): endpoint type definitions

### PR DESCRIPTION
**Pull request by drochetti**
https://github.com/user-attachments/assets/215816b3-8952-4fb8-8608-23d62aabb762

### CODE REVIEW

File: libs/client/src/client.ts [hunk 1]
```diff
@@ -44,10 +45,10 @@ export interface FalClient {
...
    * @returns the remote function output
    */
-  run<Output = any, Input = Record<string, any>>(
-    endpointId: string,
-    options: RunOptions<Input>,
-  ): Promise<Result<Output>>;
+  run<Id extends EndpointType>(
+    endpointId: Id,
+    options: RunOptions<InputType<Id>>,
+  ): Promise<Result<OutputType<Id>>>;
```
Comment (aykutkardas) [inline libs/client/src/client.ts:51]: 😍

**Review by aykutkardas — APPROVED** [review]

**Review by mederka — APPROVED** [review]
image

😭

**Comment by aykutkardas** [comment]
> image
> 😭

It will be worth it. 🙃

**Comment by drochetti** [comment]
> image
> 
> 😭 

haha don't worry, it's all generated 🤖

**Pull request merged**